drivers/wireless/cyw43/runner.zig

```diff
@@ -0,0 +1,308 @@
+                this.chip_log_state.buf[this.chip_log_state.buf_count] = b;
+                this.chip_log_state.buf_count += 1;
+            }
+
+            this.chip_log_state.last_idx += 1;
+            if (this.chip_log_state.last_idx == 1024) {
+                this.chip_log_state.last_idx = 0;
+            }
+        }
+    }
+
+    pub fn run(this: *Self) void {
+        while (true) {
+            this.log_read();
+        }
+    }
+};
+
+pub const Chip = struct {
+    arm_core_base_address: u32,
+    socsram_base_address: u32,
+    bluetooth_base_address: u32,
+    socsram_wrapper_base_address: u32,
+    sdiod_core_base_address: u32,
+    pmu_base_address: u32,
+    chip_ram_size: u32,
+    atcm_ram_base_address: u32,
+    socram_srmem_size: u32,
+    chanspec_band_mask: u32,
+    chanspec_band_2g: u32,
+    chanspec_band_5g: u32,
+    chanspec_band_shift: u32,
+    chanspec_bw_10: u32,
+    chanspec_bw_20: u32,
+    chanspec_bw_40: u32,
+    chanspec_bw_mask: u32,
+    chanspec_bw_shift: u32,
+    chanspec_ctl_sb_lower: u32,
+    chanspec_ctl_sb_upper: u32,
+    chanspec_ctl_sb_none: u32,
+    chanspec_ctl_sb_mask: u32,
+};
+
+const WRAPPER_REGISTER_OFFSET: u32 = 0x100000;
+
+const CYW43439_Chip: Chip = .{
+    .arm_core_base_address = 0x18003000 + WRAPPER_REGISTER_OFFSET,
+    .socsram_base_address = 0x18004000,
+    .bluetooth_base_address = 0x19000000,
+    .socsram_wrapper_base_address = 0x18004000 + WRAPPER_REGISTER_OFFSET,
+    .sdiod_core_base_address = 0x18002000,
+    .pmu_base_address = 0x18000000,
+    .chip_ram_size = 512 * 1024,
+    .atcm_ram_base_address = 0,
+    .socram_srmem_size = 64 * 1024,
+    .chanspec_band_mask = 0xc000,
+    .chanspec_band_2g = 0x0000,
+    .chanspec_band_5g = 0xc000,
+    .chanspec_band_shift = 14,
+    .chanspec_bw_10 = 0x0800,
+    .chanspec_bw_20 = 0x1000,
+    .chanspec_bw_40 = 0x1800,
+    .chanspec_bw_mask = 0x3800,
+    .chanspec_bw_shift = 11,
+    .chanspec_ctl_sb_lower = 0x0000,
+    .chanspec_ctl_sb_upper = 0x0100,
+    .chanspec_ctl_sb_none = 0x0000,
+    .chanspec_ctl_sb_mask = 0x0700,
+};
+
+const LogState = struct {
+    addr: u32 = 0,
+    last_idx: usize = 0,
+    buf: [256]u8 = undefined,
+    buf_count: usize = 0,
+};
+
+const Core = enum(u2) {
+    wlan = 0,
+    socram = 1,
+    sdiod = 2,
+
+    fn base_addr(this: Core) u32 {
+        return switch (this) {
+            .wlan => CYW43439_Chip.arm_core_base_address,
+            .socram => CYW43439_Chip.socsram_wrapper_base_address,
+            .sdiod => CYW43439_Chip.sdiod_core_base_address
+        };
+    }
+};
+
+pub const SharedMemData = extern struct {
+    flags: u32,
+    trap_addr: u32,
+    assert_exp_addr: u32,
+    assert_file_addr: u32,
+    assert_line: u32,
+    console_addr: u32,
+    msgtrace_addr: u32,
+    fwid: u32,
+};
+
+pub const SharedMemLog = extern struct {
+    buf: u32,
+    buf_size: u32,
+    idx: u32,
+    out_idx: u32,
+};
```

Comment:
do you have zig fmt? A bunch of these files will have newlines added by anyone who touches it, which can be annoying.